lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -969,6 +979,27 @@ bdev_nvme_queued_done(void *ref, const struct spdk_nvme_cpl *cpl)
 }
 
 static void
+bdev_nvme_admin_passthru_completion(void *arg1, void *arg2)
+{
+	struct spdk_bdev_io *bdev_io = arg1;
+
+	spdk_bdev_io_complete_nvme_status(bdev_io,
+					  bdev_io->u.nvme_admin_passthru.cpl.status.sct,
+					  bdev_io->u.nvme_admin_passthru.cpl.status.sc);
+}
+
+static void
+bdev_nvme_admin_passthru_done(void *ref, const struct spdk_nvme_cpl *cpl)
+{
+	struct spdk_bdev_io *bdev_io = spdk_bdev_io_from_ctx((struct nvme_bdev_io *)ref);
+
+	memcpy(&bdev_io->u.nvme_admin_passthru.cpl, cpl, sizeof(*cpl));
+
+	spdk_event_call(spdk_event_allocate(bdev_io->u.nvme_admin_passthru.p_lcore,
```

Comment:
While unlikely, spdk_event_allocate can actually fail in out of memory scenarios. I think you probably want to allocate the event at submission time and store a pointer to it in struct nvme_bdev_io. That way if the allocation fails you can immediately return to the user without executing the command.